plugins/unifying/fu-plugin-unifying.c

```diff
@@ -181,21 +181,53 @@ fu_plugin_update_detach (FuPlugin *plugin, FuDevice *dev, GError **error)
 	/* wait for device to come back */
 	fu_device_set_status (dev, FWUPD_STATUS_DEVICE_RESTART);
 	if (lu_device_has_flag (device, LU_DEVICE_FLAG_DETACH_WILL_REPLUG)) {
-		g_debug ("doing detach in idle");
-		g_idle_add_full (G_PRIORITY_DEFAULT_IDLE,
-				 fu_plugin_unifying_detach_cb,
-				 g_object_ref (device),
-				 (GDestroyNotify) g_object_unref);
-		if (!lu_context_wait_for_replug (data->ctx,
-						 device,
-						 FU_DEVICE_TIMEOUT_REPLUG,
-						 error))
-			return FALSE;
+		if (LU_IS_DEVICE_PERIPHERAL (device)) {
+			if (!lu_device_detach (device, error))
+				return FALSE;
+
+			if (!lu_context_wait_for_peripheral_replug (data->ctx,
```

Comment:
Why can't peripheral devices use `lu_context_wait_for_replug`?

Reply:
The unifying peripherals are not USB devices per say. When they're rebooted they don't trigger the added/removed udev events. So basically if a peripheral with the DETACH_WILL_REPLUG flag, using the usb replug will result in a timeout everytime since no usb events are triggered.